dkpro-core-io-nyt-asl/src/main/java/de/tudarmstadt/ukp/dkpro/core/io/nyt/NYTCollectionReader.java

```diff
@@ -0,0 +1,171 @@
+package de.tudarmstadt.ukp.dkpro.core.io.nyt;
+
+import java.io.IOException;
+import java.net.URL;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+
+import javax.xml.parsers.ParserConfigurationException;
+
+import org.apache.uima.UimaContext;
+import org.apache.uima.cas.CAS;
+import org.apache.uima.cas.CASException;
+import org.apache.uima.collection.CollectionException;
+import org.apache.uima.fit.component.CasCollectionReader_ImplBase;
+import org.apache.uima.fit.descriptor.ConfigurationParameter;
+import org.apache.uima.jcas.JCas;
+import org.apache.uima.jcas.cas.StringArray;
+import org.apache.uima.resource.ResourceInitializationException;
+import org.apache.uima.util.Level;
+import org.apache.uima.util.Logger;
+import org.apache.uima.util.Progress;
+import org.apache.uima.util.ProgressImpl;
+
+import com.nytlabs.corpus.NYTCorpusDocument;
+
+import de.tudarmstadt.ukp.dkpro.io.nyt.metadata.NYTArticleMetaData;
+import de.tudarmstadt.ukp.dkpro.core.api.metadata.type.DocumentMetaData;
+
+public class NYTCollectionReader extends CasCollectionReader_ImplBase {
+
+	private Logger Logger;
+
+	/**
+	 * Path to the corpus' data directory.
+	 */
+	public static final String PARAM_DATA_PATH = "dataPath";
+	@ConfigurationParameter(name = PARAM_DATA_PATH, mandatory = true)
+	private String dataPathString;
```

Comment:
The parameter for the location from which a component reads must be named PARAM_LOCATION and must be declared in this way:

```
    /**
     * Location from which the input is read.
     */
    public static final String PARAM_SOURCE_LOCATION = ComponentParameters.PARAM_SOURCE_LOCATION;
    @ConfigurationParameter(name = PARAM_SOURCE_LOCATION, mandatory = false)
    private String sourceLocation;
```